src/client.coffee

```diff
@@ -40,20 +35,41 @@ class Client
   setHost: (host) ->
     @_host = host
 
+  # Deprecated. Use addFilter.
   addContext: (context) ->
-    merge(@_context, context)
+    console?.warn?('airbrake: addContext is deprecated, please use addFilter')
+    @addFilter (notice) ->
+      notice.context = merge({}, context, notice.context)
+      return notice
 
+  # Deprecated. Use addFilter.
   setEnvironmentName: (envName) ->
-    @_context.environment = envName
+    console?.warn?('airbrake: setEnvironmentName is deprecated, please use addFilter')
+    @addFilter (notice) ->
+      if not notice.context.environment?
```

Comment:
Right, though my eyes are untrained to read `unless`/`until` and I almost never use them...